File: Core/Entities/Recipe.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Core.Entities
{
    public class Recipe
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public required string Description { get; set; }
        public required string CookingTime { get; set; }
        public required string Difficulty { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public required string ImageUrl { get; set; }

        // public ICollection<Image> Images { get; set; } = [];
        public ICollection<Category> Categories { get; set; } = [];
        public ICollection<Instruction> Instructions { get; set; } = [];
        public ICollection<Ingredient> Ingredients { get; set; } = [];
        public ICollection<Comment> Comments { get; set; } = [];
        public ICollection<Rating> Ratings { get; set; } = [];
        public string? UserId { get; set; } // Foreign key
        public AppUser? User { get; set; } // Navigation property

        public ICollection<AppUser> FavoritedBy { get; set; } = [];
    }
}

File: client/src/types/recipe.ts
export type Recipe = {
  id: number;
  name: string;
  description: string;
  cookingTime: string;
  difficulty: string;
  servingSize: number;
  averageRating: number;
  ratingCount: number;
  imageUrl: string;
  createdAt: string;
  categories: Category[];
  ingredients: Ingredient[];
  instructions: Instruction[];
};

export type RecipeCreate = {
  name: string;
  description: string;
  cookingTime: string;
  difficulty: string;
  servingSize: string;
  imageUrl: string;
};


export type Ingredient = {
  id?: number;
  name: string;
  quantity: string;
  unit: string;
  recipeId?: number;
};

export type Instruction = {
  id?: number;
  text: string;
  recipeId?: number;
};


export type Category = {
  id: number;
  name: string;
  slug: string;
}

export type AddIngredient = {
  name: string;
  quantity: string;
  unit: string;
}

export type AddInstruction = {
  text: string;
}

export type User = {
  id?: string;
  userName?: string;
  profileName?: string;
}

export type RecipeComment = {
  id: number;
  content: string;
  createdAt: string;
  userId: string;
  user: User;
}

export type RecipeUpdate = {
  name: string;
  description: string;
  cookingTime: string;
  difficulty: string;
  servingSize: number;
  imageUrl: string;
  categories: []
  instructions: []
  ingredients: []
};